### ISSUE TITLE
btcpay server magento 2 plugin

### ISSUE THREAD

**Issue by CreamyG31337** [btcpayserver/btcpayserver-doc#14]
Although someone has forked it and listed it as an option, the fork doesn't appear to contain even a simple search and replace to change bitpay to btcpayserver. Additionally, there is no issue tracker enabled on this fork. It would be nice to see at least one commit on this "fork". It's basically useless right now -- it's 6 commits behind and 0 ahead of the original... and not even a document suggesting what the purpose of it is.

**Comment by NicolasDorier** [comment]
Hey @ankitdottech any news about it? 
That said the magento2 plugin of bitpay might work out of the box on BTCPay. But I need to check this.

**Comment by CreamyG31337** [comment]
I assumed that somewhere in the code there are some bitpay invoice/checkout URLs or API endpoints or something that will need updating to point to my btcpayserver, but I didn't notice any yet. My server is almost done syncing so I can test it out soon.

**Comment by CreamyG31337** [comment]
Okay I've exposed some settings to enter a custom network in [my fork](https://github.com/CreamyG31337/magento2-plugin).
![image](https://user-images.githubusercontent.com/11222932/40580704-6c63e2e2-6101-11e8-8c1b-f1d183ed8ac5.png)
I'll try to improve it a bit more before doing any pull request. Port number should be validated as integer and that type of thing. Need to fix that policies error still before we can test it properly.

**Comment by NicolasDorier** [comment]
do you want I replace the current btcpayserver's fork with your repo and give you admin access?

**Comment by NicolasDorier** [comment]
just ping me when you are done, I will just remove the current repo and replace by yours.

**Comment by Code0Zero** [comment]
is it ready for production environment?

**Comment by CreamyG31337** [comment]
I think my fork was working but I never tested it enough to be sure. If you find some problem let me know. I'll get back to it soon.

**Comment by NicolasDorier** [comment]
@CreamyG31337 , should I replace https://github.com/btcpayserver/magento2-plugin by yours?

**Comment by CreamyG31337** [comment]
Yes it looked like it was working so I stopped but if there are more issues I'll try to fix it. Just have been busy the last month.

**Comment by NicolasDorier** [comment]
@CreamyG31337 can you transfer your repo to btcpayserver ? I delete the current magento one.

**Comment by bitcoinshirt** [comment]
It appears this has been resolved.

**Comment by Kukks** [comment]
Hey @CreamyG31337  I also forked this one a while back and tried to do some changes but the person requesting it disappeared and could not test it. https://github.com/btcpayserver/magento-plugin

Is there a difference? between magento20plugin and magento-plugin?

**Comment by stifboy** [comment]
> Hey @CreamyG31337 I also forked this one a while back and tried to do some changes but the person requesting it disappeared and could not test it. https://github.com/btcpayserver/magento-plugin
> 
> Is there a difference? between magento20plugin and magent

> Okay I've exposed some settings to enter a custom network in [my fork](https://github.com/CreamyG31337/magento2-plugin).
> ![image](https://user-images.githubusercontent.com/11222932/40580704-6c63e2e2-6101-11e8-8c1b-f1d183ed8ac5.png)
> I'll try to improve it a bit more before doing any pull request. Port number should be validated as integer and that type of thing. Need to fix that policies error still before we can test it properly.

Did you test this? I have setup your this extension but it does not seem to work. I can't pair to Btcpay server.

**Comment by CreamyG31337** [comment]
It was definitely pairing! You can see my last commit was to fix some stuff that was logging the transaction -- it was getting that far at least. I'll try to find time to look at this again soon.

**Comment by stifboy** [comment]
> It was definitely pairing! You can see my last commit was to fix some stuff that was logging the transaction -- it was getting that far at least. I'll try to find time to look at this again soon. Looks like my web host (friend's server) was erased so I need to set everything up again :(

Ok. I can provide you my server if you need. Just let me know. Thanks

**Comment by NicolasDorier** [comment]
@stifboy can you open a new issue and link it here?

**Comment by stifboy** [comment]
> @stifboy can you open a new issue and link it here?

I have opened a new issue [here](https://github.com/btcpayserver/btcpayserver-doc/issues/115)

**Comment by Kukks** [comment]
@stifboy @NicolasDorier  There already is an issue https://github.com/CreamyG31337/magento2-plugin/issues/1
The plugin is not part of the btcpayserver org for some reason

**Comment by NicolasDorier** [comment]
@CreamyG31337 do you want to transfer to https://github.com/btcpayserver/magento2-plugin ? I think you have the rigths
